apps/openmw/mwworld/esmstore.cpp

```diff
@@ -29,6 +29,7 @@ namespace
 
     void readRefs(const ESM::Cell& cell, std::vector<Ref>& refs, std::vector<std::string>& refIDs, std::vector<ESM::ESMReader>& readers)
     {
+        // TODO: we have many near identical copies of this code.
```

Comment:
They are not identical but similar.

Reply:
I suppose the definition of similarity depends on how we refactor such code and how we could manage to consolidate optimisations. Templates or visitor pattern may be helpful here. Should I change "near identical" to "similar" anyway?